shared/gh/js/views/tenant-student/gh.terms-and-conditions.js

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright 2015 Digital Services, University of Cambridge Licensed
+ * under the Educational Community License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['gh.core', 'gh.constants', 'moment'], function(gh, constants, moment) {
+
+    /**
+     * Accept the terms and conditions
+     *
+     * @private
+     */
+    var acceptTermsAndConditions = function() {
+        // Accept the terms and condtions
+        gh.api.userAPI.acceptTermsAndConditions(gh.data.me.id, function(err, data) {
+            if (err) {
+                return gh.utils.notification('Could not accept the terms and conditions', constants.messaging.default.error, 'error');
+            }
+
+            // Close the modal window
+            $('#gh-terms-and-conditions-modal').modal('hide');
+        });
+    };
+
+    /**
+     * Check the terms and conditions
+     *
+     * @private
+     */
+    var checkTermsAndConditions = function() {
+        // The terms and conditions should only be verified in they're enabled and for authenticated users
+        if (gh.config.termsAndConditionsEnabled && !gh.data.me.anon) {
```

Comment:
The `needsToAccept` should do all the checks for you. This function can just be:

```
var checkTermsAndConditions = function() {
    if (gh.data.me.termsAndConditions.needsToAccept) {
        return setupTermsAndConditionsModal();
    ]
};
```